New package: DigitalAssetExchangeFormatIO v1.0.0

JuliaRegistrator opened the pull request:
- Registering package: DigitalAssetExchangeFormatIO
- Repository: https://github.com/Larbino1/DigitalAssetExchangeFormatIO.jl
- Created by: @Larbino1
- Version: v1.0.0
- Commit: fea5991fa898823bbc2181e08559acde2e60e62e
- Reviewed by: @Larbino1
- Reference: https://github.com/Larbino1/DigitalAssetExchangeFormatIO.jl/issues/1#issuecomment-2263472757
<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a835983be2b031f89474918468535889a498e8013a990050cd0536329fec9e46a2167172826dbeb67e07f9d97dc543ddeb3ae61a9a8a1b0fb2f19f1e37a3f51f6c4df964958dc0b296ea1c371d0b0dfa68c5970d77ded76e5148cd09d746e704d0a4ae370f0f278441defa937e09f6fd0cebaa9b7ce9736cf9cb709a33e37d4520ce260bac520a0e99f58458553abee244fab1 -->

Larbino1 commented:
@rdeits @tkoolen 
Hi both, you might be interested in this - a simple dae parser for robot visualization. Once parsed into GeometryBasics meshes and colours, I can plot with GLMakie.

I would appreciate if you could lend some other files to test on, for example from the ur5 in this thread:  https://discourse.julialang.org/t/how-to-load-urdf-files-with-dae-models-into-meshcat/19509

JuliaRegistrator committed:
New package: DigitalAssetExchangeFormatIO v1.0.0

UUID: 43182933-f65b-495a-9e05-4d939cea427d
Repo: https://github.com/Larbino1/DigitalAssetExchangeFormatIO.jl.git
Tree: c2268ec34261afb616f669488eaf3cc7e5789d48

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

ffreyer commented:
[noblock] The reason for this is ultimately how OpenGL (and probably the other rendering APIs) work. You only get one index buffer per draw call, with one index per vertex. So you are forced to use the same vertex index for positions, normals, etc.

We are currently planning to restart https://github.com/JuliaGeometry/GeometryBasics.jl/pull/173 and refactor GeometryBasics. I'll probably move a function that does this from MeshIO to GeometryBasics with that. We could also consider adding something like `MultiFace` so that you can have a mesh that does not fit the OpenGL format, and then add some functionality to convert to it later.

Larbino1 commented:
> [noblock] The reason for this is ultimately how OpenGL (and probably the other rendering APIs) work. You only get one index buffer per draw call, with one index per vertex. So you are forced to use the same vertex index for positions, normals, etc.
> 
> We are currently planning to restart [JuliaGeometry/GeometryBasics.jl#173](https://github.com/JuliaGeometry/GeometryBasics.jl/pull/173) and refactor GeometryBasics. I'll probably move a function that does this from MeshIO to GeometryBasics with that. We could also consider adding something like `MultiFace` so that you can have a mesh that does not fit the OpenGL format, and then add some functionality to convert to it later.

[noblock] Hi, thanks for the reply! That sounds like a great piece of work, which will really help with the Mesh ecosystem in julia for plotting.
